posydon/visualization/plot_pop.py

```diff
@@ -251,4 +258,137 @@ def plot_hist_properties(x, df_intrinsic=None, df_observable=None,
     if path:
         plt.savefig(path)
     if show:
-        plt.show()
\ No newline at end of file
+        plt.show()
+        
+        
+def plot_popsyn_over_grid_slice(pop, grid_type, met_Zsun, slices=None, channel=None, 
+                                plot_dir='./', prop=None, prop_range=None, 
+                                log_prop=False, alpha=0.3, s=5., 
+                                show_fig=True, save_fig=True, close_fig=True,
+                                verbose=False):
+
+    # load grid
+    met = convert_metallicity_to_string(met_Zsun)
+    grid_path = os.path.join(PATH_TO_POSYDON_DATA, f'POSYDON_data/{grid_type}/{met}_Zsun.h5')
+    grid = PSyGrid(verbose=False)
+    grid.load(grid_path)
+
+    # check if formation channel information is avaialbe
+    if channel is not None and 'channel' not in pop.df_oneline.keys():
+        if verbose:
+            print('Computing formation channels...')
+        pop.get_formation_channels()
+        
+    if 'CO' in grid_path:
+        # compact object mass slices
+        def log_range(x_min,x_max,x_n):
+            return 10**np.linspace(np.log10(x_min),np.log10(x_max), x_n)
+        m_CO_min = 1.
+        m_CO_max = 51.32759630397827
+        m_CO_n = 23
```

Comment:
Why are those values for the range hard coded? Wouldn't it be better to have them as arguments with default values like those ones, here?

Reply:
fixed. I am now reading this information from the grid.intial_values